### ISSUE TITLE
Allow to Deactivate / Activate Tags instead of delete them

### ISSUE THREAD

**Issue by scysys** [matomo-org/tag-manager#23]

**Comment by tsteur** [comment]
FYI: Current work around would be to set start date to a date in the past.

**Comment by scysys** [comment]
But its not the same :)

**Comment by mattab** [comment]
(feedback from customer)

Is it possible to perhaps have a on/off option for the tags?

Currently I can only turn a tag off by going to the tag in question -> advanced options -> and set a start date in the future to delay it


Furthermore, once a tag is paused I cannot see that it is paused - only when I again go to the tag -> advanced options.

Can we maybe add something next to the tags to indicate whether they are off/on? Something similar to what we have for versions when they are live/not live

**Comment by atom-box** [comment]
_A user comment:_  

"It would be interesting to have the possibility to make some tags inactive, to be able to send them to a "trash" and then recover them. For example, I need to do a test on several tags without deleting them, but I can't (unless I haven't seen the feature). Otherwise MTM is cool! Good job."
Source: On-Premise

**Comment by atom-box** [comment]
Maybe this is the current workaround for this issue:
![image](https://user-images.githubusercontent.com/22874146/196198964-83167567-c17b-4f52-a49a-ffa560ed9d58.png)

**Comment by atom-box** [comment]
_(User comment)_

Feature:
Like: No
Choice: missingfeatures
Feedback:
Pause a tag so I can test other tags..
Source: On-Premise

**Comment by atom-box** [comment]
_(User comment)_ 

...audit of our tags and I would like to know if it is possible to see a faster method or way to conveniently display tags that are currently enabled or disabled in production stage. At the moment, we would have to go through each tag in every container to see its state.

**Comment by AltamashShaikh** [comment]
> _(User comment)_
> 
> ...audit of our tags and I would like to know if it is possible to see a faster method or way to conveniently display tags that are currently enabled or disabled in production stage. At the moment, we would have to go through each tag in every container to see its state.

if using startDate endDate, then a DB query would help to get the results faster

**Comment by atom-box** [comment]
> If using startDate endDate, then a DB query would help to get the results faster 

For example, to search for tags active in April of 2023 a user could try 
`mysql> select * from matomo_tagmanager_tag where idsite=1 AND end_date<= 2023-04-01 AND  start_date <= "2023-04-30 23:59:59"  ;`

In case it is convenient here is a description of the relevant table:

```
mysql> describe matomo_tagmanager_tag;
+--------------------+-----------------------+------+-----+-----------+----------------+
| Field              | Type                  | Null | Key | Default   | Extra          |
+--------------------+-----------------------+------+-----+-----------+----------------+
| idtag              | bigint(20) unsigned   | NO   | PRI | NULL      | auto_increment |
| idcontainerversion | bigint(20) unsigned   | NO   |     | NULL      |                |
| idsite             | int(11) unsigned      | NO   | MUL | NULL      |                |
| type               | varchar(50)           | NO   |     | NULL      |                |
| name               | varchar(50)           | NO   |     | NULL      |                |
| description        | varchar(1000)         | NO   |     |           |                |
| status             | varchar(10)           | NO   |     | NULL      |                |
| parameters         | mediumtext            | NO   |     | NULL      |                |
| fire_trigger_ids   | text                  | NO   |     | NULL      |                |
| block_trigger_ids  | text                  | NO   |     | NULL      |                |
| fire_limit         | varchar(20)           | NO   |     | unlimited |                |
| priority           | smallint(5) unsigned  | NO   |     | NULL      |                |
| fire_delay         | mediumint(8) unsigned | NO   |     | 0         |                |
| start_date         | datetime              | YES  |     | NULL      |                |
| end_date           | datetime              | YES  |     | NULL      |                |
| created_date       | datetime              | NO   |     | NULL      |                |
| updated_date       | datetime              | NO   |     | NULL      |                |
| deleted_date       | datetime              | YES  |     | NULL      |                |
+--------------------+-----------------------+------+-----+-----------+----------------+
18 rows in set (0.00 sec)
```

This is a workaround.  

It would still be nice to consider adding this feature.

**Comment by atom-box** [comment]
User is trying to find source of an error and this feature, turning off a MTM tag would be VERY USEFUL to them in trying to isolate their problem.

**Comment by atom-box** [comment]
This feature was requested today by a user.

> Is there a way to selectively remove one off changes...?

**Comment by jr180180** [comment]
I would find this useful

**Comment by AltamashShaikh** [comment]
> This feature was requested today by a user.
> 
> > Is there a way to selectively remove one off changes...?

@atom-box You can do this for now

1. Export the container
2. Remove the one of change
3. Publish new container
4. Import the container from the json exported in step1, now your draft state will contain the one of change and published state won't have that change

**Comment by ronan-hello** [comment]
Hi,

I think this is now **critical** as many users are using the tag manager. (issue was first open in 2018)
Can we have a simple "pause" button instead of setting custom datetime ?

Regards,
Ronan HELLO

**Comment by snake14** [comment]
Hi @ronan-hello . Thank you for bringing this up. We actually have an internal ticket that we're planning to start work on soon to add a pause button to tags. We'll likely work on that ticket in the within a month. I'll add a note to update this issue once work is done.

**Comment by ronan-hello** [comment]
Hi

Thank you so much for your reply, I'm glad that this is in your roadmap.

Regards,
Ronan HELLO

**Comment by AltamashShaikh** [comment]
Should be fixed with #836

**Comment by Chardonneaur** [comment]
Hello,
I tested the pause/enable feature available in the cloud and unless i didn't understand how it works, i have the feeling that it is not working:
https://github.com/user-attachments/assets/6d0d7083-b0df-43b6-ba35-3606e9ce5c48

**Comment by atom-box** [comment]
Is this on Cloud now, as a bleeding edge thing? 
It seems broken in Cloud. The new icon is there but when you click it nothing happens.

**Comment by AltamashShaikh** [comment]
@Chardonneaur @atom-box 
I just tested on my cloud instance and this seems to work for me


https://github.com/user-attachments/assets/a70ea12e-b51b-456a-a5e7-6e841dcea51f

**Comment by AltamashShaikh** [comment]
@atom-box Can you explain what broken for you ?

**Comment by AltamashShaikh** [comment]
@Chardonneaur I was able to reproduce this, will add a fix tomorrow

**Comment by AltamashShaikh** [comment]
Created https://github.com/matomo-org/tag-manager/pull/860 to fix the issue

**Comment by fritzgithub** [comment]
Hi, that would be a great feature, will it be released soon?

**Comment by AltamashShaikh** [comment]
@fritzgithub This is already live, which Matomo version are you using ?

**Comment by fritzgithub** [comment]
Thank you for the quick reply. We are using  Matomo-Version: 5.1.2.

**Comment by AltamashShaikh** [comment]
@fritzgithub we released 5.2.0-rc1 - https://github.com/matomo-org/matomo/releases/tag/5.2.0-rc1 last week, so if all good with rc and no issue is reported, 5.2.0 will be released soon with the fix.

**Comment by fritzgithub** [comment]
Thank you, 'soon' sounds good!
